spec/std/oauth2/client_spec.cr

```diff
@@ -1,6 +1,20 @@
 require "spec"
 require "oauth2"
 
+private def assert_request
+  server = HTTP::Server.new do |context|
+    body = context.request.body.not_nil!.gets_to_end
```

Comment:
Hm... this `not_nil!` here looks ugly. This has nothing to do with this PR, but I wonder how we ended up with `body` begin nilable. I mean, it makes sense, a request might not have a body. But maybe we should have `body?` (returns `nil` if none) and `body` (raises if none) as we do in other places. That way accessing the body is an assertion itself. Thoughts? (for another discussion)

Reply:
Alternatively, `body` could always be an IO, but could be an empty one. I don't think the HTTP specification differentiates between "no body" and "empty body".

Reply:
Yes, that could work too. We'd need some `IO::Empty` or something...